KeenClient/KeenClient.m

```diff
@@ -354,6 +354,11 @@ + (KeenClient *)sharedClientWithProjectID:(NSString *)projectID
         return nil;
     }
 
+    // If the keys are already set, return the existing instance, don't overwrite the keys
+    if (self.sharedClient.projectID || self.sharedClient.writeKey || self.sharedClient.readKey) {
+        return self.sharedClient;
```

Comment:
Hmm... yeah that does sound like it could be a valid use-case, so seems fine to return the sharedClient instance.

I'm not sure how I feel about this singleton pattern in general since there's a required initialization step which is coupled with access, and the semantic coupling that introduces. I.e. calling sharedClientWithProjectID: before calling sharedClient:... I'd argue it might be less bug-prone to require clients to call initWithProjectID: on an instance and use that. But that's maybe a topic for another day.